### ISSUE TITLE
StackOverflowError caused by two yield calls

### ISSUE THREAD

**Issue by chemist777** [vert-x3/vertx-lang-kotlin#233]
### Version

Which version(s) did you encounter this bug ?

vertx-lang-kotlin-coroutines 4.4.2
kotlinx-coroutines-core: 1.6.1

### Context

I encountered an exception while writing two consecutive `yield` calls in a CoroutineVerticle.

### Do you have a reproducer?

A reproducer:

```kotlin
@Test
fun bug(): Unit = runBlocking {
    // it's a workaround to show correct exceptions instead of NoClassDefFoundError
    Class.forName("kotlinx.coroutines.CoroutineExceptionHandlerImplKt")
    Class.forName("kotlin.internal.PlatformImplementationsKt")
    // actual reproducer
    val vertx = Vertx.vertx()
    val dispatcher = vertx.orCreateContext.dispatcher()
    val scope = CoroutineScope(dispatcher + SupervisorJob())
    scope.async {
        (0 until 1000).forEach {
            yield()
            yield()
        }
    }.await()
}
```

### Steps to reproduce

1. Run the snippet.
2. Observe a hang and the following StackOverflowError:
```
Exception in thread "vert.x-eventloop-thread-0 @coroutine#2" java.lang.RuntimeException: Exception while trying to handle coroutine exception
	at kotlinx.coroutines.CoroutineExceptionHandlerKt.handlerException(CoroutineExceptionHandler.kt:38)
	at kotlinx.coroutines.CoroutineExceptionHandlerKt.handleCoroutineException(CoroutineExceptionHandler.kt:29)
	at kotlinx.coroutines.DispatchedTask.handleFatalException(DispatchedTask.kt:146)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:115)
	at io.vertx.kotlin.coroutines.VertxCoroutineExecutor.execute(VertxCoroutine.kt:218)
	at kotlinx.coroutines.ExecutorCoroutineDispatcherImpl.dispatch(Executors.kt:128)
	at kotlinx.coroutines.CoroutineDispatcher.dispatchYield(CoroutineDispatcher.kt:150)
	at kotlinx.coroutines.internal.DispatchedContinuation.dispatchYield$kotlinx_coroutines_core(DispatchedContinuation.kt:264)
	at kotlinx.coroutines.YieldKt.yield(Yield.kt:37)
	at io.coinmetrics.httpserver.ProcessingPoolTest$bug$1$1.invokeSuspend(ProcessingPoolTest.kt:115)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at io.vertx.kotlin.coroutines.VertxCoroutineExecutor.execute(VertxCoroutine.kt:218)
	at kotlinx.coroutines.ExecutorCoroutineDispatcherImpl.dispatch(Executors.kt:128)
	at kotlinx.coroutines.CoroutineDispatcher.dispatchYield(CoroutineDispatcher.kt:150)
	at kotlinx.coroutines.internal.DispatchedContinuation.dispatchYield$kotlinx_coroutines_core(DispatchedContinuation.kt:264)
	at kotlinx.coroutines.YieldKt.yield(Yield.kt:37)
	at io.coinmetrics.httpserver.ProcessingPoolTest$bug$1$1.invokeSuspend(ProcessingPoolTest.kt:114)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at io.vertx.kotlin.coroutines.VertxCoroutineExecutor.execute(VertxCoroutine.kt:218)
	at kotlinx.coroutines.ExecutorCoroutineDispatcherImpl.dispatch(Executors.kt:128)
	at kotlinx.coroutines.CoroutineDispatcher.dispatchYield(CoroutineDispatcher.kt:150)
	at kotlinx.coroutines.internal.DispatchedContinuation.dispatchYield$kotlinx_coroutines_core(DispatchedContinuation.kt:264)
	at kotlinx.coroutines.YieldKt.yield(Yield.kt:37)
	at io.coinmetrics.httpserver.ProcessingPoolTest$bug$1$1.invokeSuspend(ProcessingPoolTest.kt:115)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at io.vertx.kotlin.coroutines.VertxCoroutineExecutor.execute(VertxCoroutine.kt:218)
	at kotlinx.coroutines.ExecutorCoroutineDispatcherImpl.dispatch(Executors.kt:128)
	at kotlinx.coroutines.CoroutineDispatcher.dispatchYield(CoroutineDispatcher.kt:150)
	at kotlinx.coroutines.internal.DispatchedContinuation.dispatchYield$kotlinx_coroutines_core(DispatchedContinuation.kt:264)
	at kotlinx.coroutines.YieldKt.yield(Yield.kt:37)
	at io.coinmetrics.httpserver.ProcessingPoolTest$bug$1$1.invokeSuspend(ProcessingPoolTest.kt:114)
        ... and so on (a recursion) 
```

Most likely, the error is caused by the following fragment in VertxCoroutine.kt:
```kotlin
  override fun execute(command: Runnable) {
    if (Vertx.currentContext() != vertxContext) {
      vertxContext.runOnContext { command.run() }
    } else {
      command.run()
    }
  }
```

### Extra
Java 11

**Comment by vietj** [comment]
what does happen with a regular kotlin coroutine ?

**Comment by chemist777** [comment]
> what does happen with a regular kotlin coroutine ?

The following code works well with regular thread pools. The problem is related to the thread pool managed by Vert.x/vertx-lang-kotlin.

```kotlin
@Test
fun `it works well with standard dispatchers`(): Unit = runBlocking {
    // it's a workaround to show correct exceptions instead of NoClassDefFoundError
    Class.forName("kotlinx.coroutines.CoroutineExceptionHandlerImplKt")
    Class.forName("kotlin.internal.PlatformImplementationsKt")
    // actual reproducer
    val dispatcher = Executors.newFixedThreadPool(4).asCoroutineDispatcher()
    val scope = CoroutineScope(dispatcher + SupervisorJob())
    scope.async {
        (0 until 1000).forEach {
            yield()
            yield()
        }
    }.await()
}
```

**Comment by tsegismont** [comment]
Fixed by #242
